frontend/src/scenes/insights/utils.ts

```diff
@@ -19,3 +19,12 @@ export const getLogicFromInsight = (
         return trendsLogic(logicProps)
     }
 }
+
+export const getDisplayNameFromEntityFilter = (
+    filter: EntityFilter | ActionFilter | null,
+    isCustom = true
+): string | null => {
+    const customName = (filter?.custom_name ?? '').trim() === '' ? null : filter?.custom_name
```

Comment:
Nit: this took me a bit to understand what was happening. Might be good to break it out into a few lines.